### PR TITLE
Add notes on v5.x common pitfalls.

### PR DESCRIPTION
**Pull request by remeh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,19 @@ call the method `Datadog::Statsd#flush` if you want to force the sending of metr
...
   * properly closing your DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close` to release the resources used by the instance and to close the socket
 
+### Known incompatibility with v5.x
+
+Version v5.x of DogStatsD-ruby is using a companion thread for preemptive flushing, it brings better performances for application having a high-throughput of statsd metrics, but it comes with incompatibility with:
+
+    * applications forking after having created the dogstatsd instance: forking a process can't duplicate the existing threads, meaning that one of the process won't have a companion thread to flush the metrics
+    * applications creating a lot of different instances of the client without closing them. It is important to close the instance to free the thread and the socket it is using
+
+If you are using [Sidekiq](https://github.com/mperham/sidekiq), please make sure to close the client instance that are instanciated. [See this example on using DogStatsD-ruby v5.x with Sidekiq](https://github.com/DataDog/dogstatsd-ruby/blob/master/examples/sidekiq_example.rb)
```
Comment (sgnn7) [inline README.md:64]:
```suggestion
If you are using [Sidekiq](https://github.com/mperham/sidekiq), please make sure to close the client instance that are instantiated. [See this example on using DogStatsD-ruby v5.x with Sidekiq](https://github.com/DataDog/dogstatsd-ruby/blob/master/examples/sidekiq_example.rb).
```

File: README.md [hunk 2]
```diff
@@ -54,6 +54,19 @@ call the method `Datadog::Statsd#flush` if you want to force the sending of metr
   * using singletons instances of the DogStatsD client and not allocating one each time you need one, letting the buffering mechanism flush metrics, or,
   * properly closing your DogStatsD client instance when it is not needed anymore using the method `Datadog::Statsd#close` to release the resources used by the instance and to close the socket
 
+### Known incompatibility with v5.x
```
Comment (sgnn7) [inline README.md:57]: These aren't technically incompatibilities. Maybe have the title be "v5.0 Common Pitfalls" or something like that?

File: README.md [hunk 3]
```diff
@@ -54,6 +54,19 @@ call the method `Datadog::Statsd#flush` if you want to force the sending of metr
...
+
+Version v5.x of DogStatsD-ruby is using a companion thread for preemptive flushing, it brings better performances for application having a high-throughput of statsd metrics, but it comes with incompatibility with:
```
Comment (sgnn7) [inline README.md:59]:
```suggestion
Version v5.x of `dogstatsd-ruby` is using a companion thread for preemptive flushing, it brings better performances for application having a high-throughput of statsd metrics, but it comes with incompatibility with:
```

**Review by sgnn7 — APPROVED** [review]
@remeh Just a few formatting nits

File: README.md (hunk 2 above)
Comment (remeh) [reply README.md:57]: Good advice, pushing this change.

**Review by sgnn7 — APPROVED** [review]

**Review by olivielpeau — APPROVED** [review]
a few nits, but lgtm thanks

**Pull request merged**